openpype/hosts/traypublisher/csv_publish.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+import pyblish.api
+import pyblish.util
+
+from openpype.client import get_asset_by_name
+from openpype.lib.attribute_definitions import FileDefItem
+from openpype.pipeline import install_host
+from openpype.pipeline.create import CreateContext
+
+from openpype.hosts.traypublisher.api import TrayPublisherHost
+
+
+def csvpublish(
+    csv_filepath,
+    project_name,
+    asset_name,
+    task_name=None,
+    username=None,
+    targets=None
+):
+    """Publish CSV file.
+
+    Args:
+        csv_filepath (str): Path to CSV file.
+        project_name (str): Project name.
+        asset_name (str): Asset name.
+        task_name (Optional[str]): Task name.
+        username (Optional[str]): User name.
+        hostname (Optional[str]): Host name.
+        targets (Optional[list[str]]): List of targets.
+    """
+
+    # initialization of host
+    host = TrayPublisherHost()
+    install_host(host)
+
+    # setting host context into project
+    host.set_project_name(project_name)
+
+    # add asset context to environment
+    # TODO: perhaps this can be done in a better way?
+    os.environ.update({
+        "AVALON_PROJECT": project_name,
+        "AVALON_ASSET": asset_name,
+        "AVALON_TASK": task_name
+    })
+
+    # form precreate data with field values
+    file_field = FileDefItem.from_paths([csv_filepath], False).pop().to_dict()
+    precreate_data = {
+        "csv_filepath_data": file_field,
+    }
+
+    # create context initialization
+    create_context = CreateContext(host, headless=True)
+    asset_doc = get_asset_by_name(
+        project_name,
+        asset_name
+    )
+
+    create_context.create(
+        "io.openpype.creators.traypublisher.csv_ingest",
+        "Main",
+        asset_doc=asset_doc,
+        task_name=task_name,
+        pre_create_data=precreate_data,
+    )
+
+    # publishing context initialization
+    pyblish_context = pyblish.api.Context()
+    pyblish_context.data["create_context"] = create_context
+
+    # setting username and targets
+    if username:
+        pyblish_context.data["user"] = username
```

Comment:
I think we should remove `username` argument.

If we would keep it, then this code won't change anything. Key "user" is always overriden with value from `get_openpype_username` during `CollectCurrentUserPype`, and that plugin cannot be disabled, because the plugin is there to bypass default pyblish plugin.

That can be affected (only in OpenPype, not in AYON) by changing env variable `"OPENPYPE_USERNAME"`.

Reply:
I agree that this is redundant. Also, I can see a collision with the Ftrack User which has to be set via env var. 
@jrsndl would this be complication for you not to include username in CLI? 
Or what is the usecase for it? 

I can see that this needs to be set as vendor name in some cases but would this need to differ form the publishing user?

Reply:
for now I am removing usernem via 0d4ad3a85227b87624a56fd1cbda0a5a26612dee